### PR TITLE
improvements: disable cell onpress and ondrag while editing, added outside click unfocus feature

### PR DESCRIPTION
**Pull request by thetaungg**

### CODE REVIEW

File: src/datagrid/plugins/cell-selection.tsx [hunk 1]
```diff
@@ -53,23 +52,21 @@ const useOnSelectCell = () => {
...
 const allowedTargetIds = ['drag-handle'];
 
-const useResetSelectionOnClickOutside = () => {
+const useOnResetSelectionOnClickOutside = () => {
     const { set: setStore } = usePluginsDataStoreRef();
 
-    const onMouseDown = useCallback(
+    return useCallback(
         (e: MouseEvent) => {
             if (allowedTargetIds.includes((e.target as HTMLDivElement)?.id)) return;
```
Comment (elvisduru) [inline src/datagrid/plugins/cell-selection.tsx:71]: Need to handle mobile

Comment (thetaungg) [reply src/datagrid/plugins/cell-selection.tsx:71]: We can't handle on mobile, yet.
This function will only run on the web. Won't throw an error on mobile

Comment (elvisduru) [reply src/datagrid/plugins/cell-selection.tsx:71]: yeah, I meant to check and return, like you were doing in the useEffect that was deleted.

Comment (thetaungg) [reply src/datagrid/plugins/cell-selection.tsx:71]: I moved it

Comment (elvisduru) [reply src/datagrid/plugins/cell-selection.tsx:71]: okay

Comment (thetaungg) [reply src/datagrid/plugins/cell-selection.tsx:71]: It's in [useHandleClickOutside.tsx](https://github.com/webbeetechnologies/bamboo-molecules/pull/249/files#diff-0cf35c807e5df1ef47ccd8f80481c13e57176226452555c849ed67f77833b818)

**Review by elvisduru — APPROVED** [review]

**Pull request merged**